examples/halo_test/exchange_test.f90

```diff
@@ -287,10 +287,8 @@ subroutine test_div_haloX()
 
       halo_extents = halo_extents_t(1, [s1, s2, s3], decomp_main, 1, global)
 
-      allocate (vh(halo_extents%xs:halo_extents%xe, &
-                   halo_extents%ys:halo_extents%ye, &
-                   halo_extents%zs:halo_extents%ze))
-      allocate (wh, mold=vh)
+      call alloc_x(vh, opt_global=global, opt_levels=[0, 1, 1])
```

Comment:
I can extend the alloc interface for a scalar opt_level. If needed ?

Reply:
I tried something similar recently with gfortran it would always expand the scalar into an array and pad it with invalid values. If you can get it to work then certainly it would be useful

Reply:
Perhaps we could open this as a new issue if it is desired?